torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -3748,7 +3748,6 @@ def _test_barrier_helper(
 
         @skip_if_no_gpu
         @sandcastle_skip_if(BACKEND == "mpi", "MPI doesn't supports GPU barrier")
-        @sandcastle_skip_if(BACKEND == "ucc", "flaky on PyTorch CI with timeout")
```

Comment:
Can we change the condition to
`@sandcastle_skip_if(BACKEND == "ucc" and IS_SANDCASTLE, "Skipped internally")`